### PR TITLE
Add pkg-config file for opencryptoki

### PR DESCRIPTION
**Pull request by ifranzki**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -682,6 +682,7 @@ AC_LANG_POP([C++])
 AC_CONFIG_MACRO_DIRS([m4])
 
 AC_CONFIG_FILES([Makefile				\
+		 opencryptoki.pc			\
```
Comment (juergenchrist) [inline configure.ac:685]: Doesn't work with `make prefix=/foo/bar` which is allowed by autotools.

**Review by juergenchrist — APPROVED** [review]

File: configure.ac (hunk 1 above)
Comment (ifranzki) [reply configure.ac:685]: So what would be the solution (if any)?

**Comment by jschmidb** [comment]
- Yes, the usual location of the pkg-config file is ${libdir}/pkgconfig. What I learned with libzpc is: It could also be /usr/share/pkgconfig or /usr/{lib,lib64}/pkgconfig.
- Question: why are you using different notations:  @libdir@ vs. ${libdir} ?
- In libzpc the pkg-config file is also referenced in the spec file: refer to https://github.com/opencryptoki/libzpc/blob/main/libzpc.spec, it's under %files devel

File: configure.ac (hunk 1 above)
Comment (juergenchrist) [reply configure.ac:685]: Either document that this is not supported or generate the .pc-file during make.  Not sure if anybody uses this feature.  Never found a use case myself and always wondered why I can configure a prefix directory and the decide during make to use a different one.  But yes, all variables (`prefix`, `libdir`, ...) can be overwritten when invoking `make`.  But to make things even worse, I think you can even overwrite them when doing a `make install`...

Summing up, I think documenting that his variable overwriting during make is not supported might be the way to go.

**Comment by ifranzki** [comment]
> Question: why are you using different notations: @libdir@ vs. ${libdir} ?

@variable@ is replaced by autoconf when processing the *.in files during configure step.
${variable} is a variable within the pkgconfig file and is resolved by pkg-config at runtime.

**Comment by ifranzki** [comment]
> In libzpc the pkg-config file is also referenced in the spec file

Good point, I will add it to the spec file as well (although the spec file is already very much out of date and does possibly not work anyway....)

**Review by jschmidb — APPROVED** [review]
ok, looks good

File: configure.ac (hunk 1 above)
Comment (ifranzki) [reply configure.ac:685]: Created issue https://github.com/opencryptoki/opencryptoki/issues/577 so that we don't forget about this.
